platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/componentframework/TaskProcessors.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.componentframework;
+
+import com.swirlds.common.threading.framework.QueueThread;
+import com.swirlds.common.threading.framework.config.QueueThreadConfiguration;
+import com.swirlds.common.threading.interrupt.InterruptableConsumer;
+import com.swirlds.common.threading.utility.MultiHandler;
+import com.swirlds.platform.componentframework.internal.ProcessorParts;
+import com.swirlds.platform.componentframework.internal.QueueSubmitter;
+import com.swirlds.platform.componentframework.internal.TaskProcessorUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * A framework for setting up and running a set of {@link TaskProcessor}s. Each task processor will run in its own
+ * thread that will continuously hand it tasks to process if available. The components that need to submit tasks should
+ * do so via the submitter created by this framework. The steps to using this framework are:
+ * <ol>
+ *     <li>Create a {@link TaskProcessors} instance, declaring the processors in the constructor</li>
+ *     <li>Request any submitter if other components have dependencies on task processors declared</li>
+ *     <li>Add implementations for processors declared in the constructor</li>
+ *     <li>Start the framework</li>
+ *     <li>Stop the framework when done</li>
+ * </ol>
+ */
+public class TaskProcessors {
```

Comment:
I wonder if we should consider another name. I know what this does, but the name is not extremely descriptive. Perhaps something like `TaskProcessorBuilder`... we should think on this and discuss.